tools/ci/utils.ps1

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Silence errors to avoid PS failing when git commands write to stderr.
+# See https://github.com/microsoft/azure-pipelines-yaml/issues/306
+function Invoke-NoFailOnStdErr($Command, $DisplayName = $Command) {
+    $oldEAPreference = $ErrorActionPreference
+    $ErrorActionPreference = "SilentlyContinue"
+
+    $out = Invoke-Expression $Command -ErrorVariable err
+
+    if ($LastExitCode -ne 0) {
+        $ErrorActionPreference = $oldEAPreference
+        $msg = "Command '$DisplayName' returned an error"
+        Write-Host $msg
+        Write-Host $out
+        Write-Host $err
+        Write-Host "##vso[task.complete result=Failed;]$msg"
+        exit 1
+    }
+    $ErrorActionPreference = $oldEAPreference
```

Comment:
Use try/catch block here in case the `$Command` throws an exception, to be sure to restore the EAP?

Reply:
It seems that try/catch won't catch errors in native command. OTOH it will catch e.g. Powershell syntax errors which seems to be silently suppressed at the moment, so worth adding.